### PR TITLE
Add RDS connectivity for MySQL and PostgreSQL

### PR DESCRIPTION
**Pull request by hunterwerlla**
- Add connection interceptor for RDS
Kapture 2020-06-10 at 14 55 27

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/rds/auth/IamAuth.kt [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+    private fun getCredentials(connection: ProtoConnection): Credentials? {
+        val authInformation = validateConnection(connection)
+        val authToken = generateAuthToken(authInformation)
+        return Credentials(authInformation.user, authToken)
+    }
+
+    internal fun validateConnection(connection: ProtoConnection): RdsAuthInformation {
+        val regionId = connection.connectionPoint.additionalJdbcProperties[REGION_ID_PROPERTY]
+            ?: throw IllegalArgumentException(message("rds.validation.no_region_specified"))
```
[the remark this answers is not in the document]
Comment (hunterwerlla) [reply jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/rds/auth/IamAuth.kt:97]: Screen Shot 2020-06-10 at 3 46 46 PM

Now bubbling them up properly

File: jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/rds/auth/IamAuthWidget.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    override fun createPanel(): JPanel {
+        val panel = JPanel(GridLayoutManager(3, 6))
+        addUserField(panel, 0)
+
+        val credsLabel = JBLabel(message("aws_connection.credentials.label"))
+        val regionLabel = JBLabel(message("aws_connection.region.label"))
+        panel.add(credsLabel, UrlPropertiesPanel.createLabelConstraints(1, 0, credsLabel.preferredSize.getWidth()))
+        panel.add(credentialSelector, UrlPropertiesPanel.createSimpleConstraints(1, 1, 3))
+        panel.add(regionLabel, UrlPropertiesPanel.createLabelConstraints(2, 0, regionLabel.preferredSize.getWidth()))
+        panel.add(regionSelector, UrlPropertiesPanel.createSimpleConstraints(2, 1, 3))
```
[the remark this answers is not in the document]
Comment (hunterwerlla) [reply jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/rds/auth/IamAuthWidget.kt:42]: Oh, also proxy's

File: jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/rds/auth/IamAuth.kt [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        return RdsAuthInformation(
+            host,
+            port,
+            user,
+            credentialProvider,
+            region
+        )
+    }
+
+    private fun generateAuthToken(authInformation: RdsAuthInformation): String {
```
Comment (abrooksv) [inline jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/rds/auth/IamAuth.kt:129]: Can we write a unit test for this?

Comment (hunterwerlla) [reply jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/rds/auth/IamAuth.kt]: I knew I forgot something, added

**Pull request merged**